src/flash/verifyFlashEncryption.ts

```diff
@@ -0,0 +1,245 @@
+  flashType: ESP.FlashType,
+  workspaceRoot: vscode.Uri
+): Promise<FlashCheckResult> {
+  Logger.info(`Using flash type: ${flashType}`, { tag: "Flash" });
+
+  try {
+    if (flashType !== ESP.FlashType.UART) {
+      const errorMessage = `Invalid flash type for partition encryption. Required: UART, Found: ${flashType}. \n Choose one of the actions presented in the top center quick pick menu and re-flash.`;
+      const error = new Error(errorMessage);
+      const customButtons = [
+        {
+          label: "Change flash type to UART",
+          action: () => {
+            idfConf.writeParameter(
+              "idf.flashType",
+              "UART",
+              vscode.ConfigurationTarget.WorkspaceFolder,
+              workspaceRoot
+            );
+            const saveMessage = vscode.l10n.t(
+              "Flashing method successfully changed to UART"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+        {
+          label: "Disable Flash Encryption",
+          action: () => {
+            disableFlashEncryption();
+            const saveMessage = vscode.l10n.t(
+              "Flash encryption has been disabled in the SDK configuration"
+            );
+            Logger.infoNotify(saveMessage);
+            OutputChannel.appendLineAndShow(saveMessage, "Flash Encryption");
+          },
+        },
+      ];
+
+      OutputChannel.appendLineAndShow(errorMessage, "Flash Encryption");
+      Logger.errorNotify(
+        errorMessage,
+        error,
+        "verifyFlashEncryption !ESP.FlashType.UART",
+        { tag: "Flash Encryption" }
+      );
+      await showQuickPickWithCustomActions(
+        "Pick one of the following actions to continue",
+        customButtons
+      );
+      return {
+        success: false,
+        resultType: FlashCheckResultType.ErrorInvalidFlashType,
+      };
+    }
+
+    const valueEncryptionEnabled = await utils.getConfigValueFromBuild(
+      "SECURE_FLASH_ENC_ENABLED",
+      workspaceRoot
+    );
+    if (!valueEncryptionEnabled) {
+      const errorMessage =
+        "Flash encryption is enabled in the SDK configuration, but the project has not been rebuilt with these settings. Please rebuild the project to apply the encryption settings before attempting to flash the device.";
+      const error = new Error(errorMessage);
+      OutputChannel.appendLineAndShow(errorMessage, "Flash Encryption");
+      Logger.errorNotify(
+        errorMessage,
+        error,
+        "verifyFlashEncryption !valueEncryptionEnabled",
+        { tag: "Flash Encryption" }
+      );
+
+      return {
+        success: false,
+        resultType: FlashCheckResultType.ErrorEncryptionArgsRequired,
+      };
+    }
+
+    const idfTarget = await getIdfTargetFromSdkconfig(workspaceRoot);
+    const eFuse = new ESPEFuseManager(workspaceRoot);
+    const data = await eFuse.readSummary();
```

Comment:
The readSummary function takes some time to load and the user doesn't see any information until it is completed. Suggestion wrapping this function into 

```
vscode.window.withProgress(
  {
    cancellable: true,
    location: ProgressLocation,
    title: "ESP-IDF: Checking encryption eFuse...",
  },
  async (
    progress: vscode.Progress<{
      message: string;
      increment: number;
    }>,
    cancelToken: vscode.CancellationToken
  ) => {}
);
```
Tested using esp32 Core board using esp-idf v5.4 with flash_encryption example.

Otherwise LGTM

Reply:
Hi @brianignacio5, good idea! I've added it. Please take another look!